syntax = "proto3";

// A reference to an object
message ObjRef {
    string ref = 1;
}

service GenomeAnnotationAPI {
    rpc get (ObjRef) returns (GenomeAnnotation) {};
    rpc get_taxon (ObjRef) returns (Taxon) {};
}

// Semantic version: major.minor[.other]
message SemVer {
    string version = 1; // all together
    int32 major = 2;
    int32 minor = 3;
    string other = 4;
}

message GenomeAnnotation {
    SemVer version = 1;
    string ident = 2;
}

message Taxon {
    SemVer version = 1;
    string ident = 2;
    message LineageItem {
        string pcode = 1; // Phylogenetic code
        string value = 2; // Value of this item
    }
    repeated LineageItem lineage_parts = 3;
    string taxid = 4;           // Taxonomic ID
    repeated string aliases = 5;
}
